File: logging/declarative-logging/declarative-logging-annotation-processor/src/test/java/ru/tinkoff/kora/logging/aspect/mdc/MDCContextHolder.java
package ru.tinkoff.kora.logging.aspect.mdc;

import ru.tinkoff.kora.logging.common.arg.StructuredArgumentWriter;

import java.util.Collections;
import java.util.Map;

public class MDCContextHolder {

    private Map<String, StructuredArgumentWriter> mdcContext;

    public Map<String, StructuredArgumentWriter> get() {
        return mdcContext;
    }

    public void set(Map<String, StructuredArgumentWriter> mdcContext) {
        this.mdcContext = mdcContext == null
            ? Collections.emptyMap()
            : Map.copyOf(mdcContext);
    }
}

File: logging/declarative-logging/declarative-logging-symbol-processor/src/test/kotlin/ru/tinkoff/kora/logging/symbol/processor/aop/mdc/MDCContextHolder.kt
package ru.tinkoff.kora.logging.symbol.processor.aop.mdc

import ru.tinkoff.kora.logging.common.arg.StructuredArgumentWriter

class MDCContextHolder {

    private var mdcContext: Map<String, StructuredArgumentWriter>? = null

    fun get(): Map<String, StructuredArgumentWriter>? {
        return mdcContext
    }

    fun set(mdcContext: Map<String, StructuredArgumentWriter>?) {
        this.mdcContext = mdcContext?.toMap() ?: emptyMap()
    }
}
